lib/__tests__/watch.test.ts

```diff
@@ -0,0 +1,191 @@
+import chokidar from 'chokidar';
+import PQueue from 'p-queue';
+
+import { uploadFolder } from '../cms/uploadFolder';
+import { logger } from '../logger';
+import { watch } from '../cms/watch';
+import { MODE } from '../../constants/files';
+
+jest.mock('chokidar');
+jest.mock('axios');
+jest.mock('p-queue');
+jest.mock('../cms/uploadFolder');
+
+describe('watch function', () => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let chokidarMock: any;
+  let pQueueAddMock: jest.Mock;
+
+  beforeEach(() => {
+    chokidarMock = {
+      watch: jest.fn().mockReturnThis(),
+      on: jest.fn().mockReturnThis(),
+    };
+    (chokidar.watch as jest.Mock).mockReturnValue(chokidarMock);
+
+    pQueueAddMock = jest.fn();
+
+    // @ts-expect-error test case
+    PQueue.mockImplementation(() => ({
+      add: pQueueAddMock,
+      size: 0,
+    }));
+    logger.log = jest.fn();
+    logger.debug = jest.fn();
+  });
+
+  it('should call chokidar.watch with correct arguments', () => {
+    const accountId = 123;
+    const src = 'src-folder';
+    const dest = 'dest-folder';
+    const options = {
+      mode: MODE.draft,
+      remove: false,
+      disableInitial: true,
+      notify: '',
+      commandOptions: {},
+      filePaths: [],
+    };
+
+    watch(accountId, src, dest, options);
+
+    expect(chokidar.watch).toHaveBeenCalledWith(src, {
+      ignoreInitial: true,
+      ignored: expect.any(Function),
+    });
+  });
+
+  it('should trigger folder upload on initialization', async () => {
+    const accountId = 123;
+    const src = 'src-folder';
+    const dest = 'dest-folder';
+    const options = {
+      mode: MODE.draft,
+      remove: false,
+      disableInitial: false,
+      notify: '',
+      commandOptions: {},
+      filePaths: [],
+    };
+    const postInitialUploadCallback = jest.fn();
+
+    (uploadFolder as jest.Mock).mockResolvedValueOnce([]);
+
+    await watch(accountId, src, dest, options, postInitialUploadCallback);
```

Comment:
I noticed that sometimes the tests `await watch` and sometimes they don't.  I gave it a quick glance, and it doesn't appear `async`, so maybe we can remove the `await` in those instances.

Reply:
I wrote an earlier iteration of these tests that was async and then realized it wasn't necessary 💀 Some awaits survived.

Reply:
Ok, ran through this again, and this test does need to be asynchronous.